core/new-gui/src/app/workspace/service/execute-workflow/execute-workflow.service.ts

```diff
@@ -0,0 +1,199 @@
+   *  to execute the workflow and get the results.
+   *
+   * @param workflowPlan
+   */
+  public executeWorkflow(): void {
+    // get the current workflow graph
+    const workflowPlan = this.workflowActionService.getTexeraGraph();
+
+    // create a Logical Plan based on the workflow graph
+    const body = ExecuteWorkflowService.getLogicalPlanRequest(workflowPlan);
+    const requestURL = `${AppSettings.getApiEndpoint()}/${EXECUTE_WORKFLOW_ENDPOINT}`;
+
+    console.log(`making http post request to backend ${requestURL}`);
+    console.log(body);
+
+    this.executeStartedStream.next('execution started');
+
+    // make a http post request to the API endpoint with the logical plan object
+    this.http.post<SuccessExecutionResult>(
+      requestURL,
+      JSON.stringify(body),
+      { headers: { 'Content-Type': 'application/json' } })
+      .subscribe(
+        // backend will either respond an execution result or an error will occur
+        // handle both cases
+        response => this.handleExecuteResult(response),
+        errorResponse => this.handleExecuteError(errorResponse)
+      );
+  }
+
+  /**
+   * Gets the observable for execution started event
+   * Contains a string that says:
+   *  - execution process has begun
+   */
+  public getExecuteStartedStream(): Observable<string> {
+    return this.executeStartedStream.asObservable();
+  }
+
+  /**
+   * Gets the observable for execution ended event
+   * If execution succeeded, it contains an object with type
+   *  `SuccessExecutionResult`:
+   *    -  resultID: the result ID of this execution
+   *    -  Code: the result code of 0
+   *    -  result: the actual result data to be displayed
+   *
+   * If execution succeeded, it contains an object with type
+   *  `ErrorExecutionResult`:
+   *    -  Code: the result code 1
+   *    -  message: error message
+   */
+  public getExecuteEndedStream(): Observable<ExecutionResult> {
+    return this.executeEndedStream.asObservable();
+  }
+
+  /**
+   * Handles valid execution result from the backend.
+   * Sends the execution result to the execution end event stream.
+   *
+   * @param response
+   */
+  private handleExecuteResult(response: SuccessExecutionResult): void {
+    console.log('handling success result ');
+    console.log(response);
+    this.executeEndedStream.next(response);
+  }
+
+  /**
+   * Handler function for invalid execution.
+   *
+   * Send the error messages generated from
+   *  backend (if workflow is invalid or server error)
+   *  or frontend (if there's no network connection)
+   *  to the execution end event stream.
+   *
+   * @param errorResponse
+   */
+  private handleExecuteError(errorResponse: HttpErrorResponse): void {
+    console.log('handling error result ');
+    console.log(errorResponse);
+
+    // error shown to the user in different error scenarios
+    const displayedErrorMessage = ExecuteWorkflowService.processErrorResponse(errorResponse);
```

Comment:
Why invoke class name and then call function? In rest of the code, local function is directly called.

Reply:
that's calling a static function, similar to Java

Reply:
When a function does not involve touching a class member, I prefer it to be a static function.

Reply:
I think you misunderstood...I was saying that the function is being called in its own class... So, the class name is optional (whether function is static or non-static)

Reply:
It's not optional in Typescript

Reply:
oh...then it's ok